src/cascade/model/age_time_grid.py

```diff
@@ -0,0 +1,153 @@
+from datetime import timedelta
+from itertools import product
+from math import nan, inf
+
+import numpy as np
+import pandas as pd
+
+from cascade.dismod.constants import PriorKindEnum
+
+GRID_SNAP_DISTANCE = 1 / timedelta(days=365).total_seconds()
+"""Times within one second are considered equal."""
+
+
+class AgeTimeGrid:
+    """The AgeTime grid holds rows of a table at each age and time value.
+
+    At each age and time point is a dataframe consisting of the columns
+    given in the constructor. So getting an item returns a dataframe
+    with those columns. Setting a dataframe sets those columns.
+    Each AgeTimeGrid has three possible mulstds, for value, dage, dtime.
+
+    >>> atg = AgeTimeGrid(([0, 10, 20], [1990, 2000, 2010]), ["height", "weight"])
+    >>> atg[:, :] = [6.1, 195]
+    >>> atg[:, :].height = [5.9]
+    >>> atg[10, 2000] = [5.7, 180]
+    >>> atg[5:17, 1980:1990].weight = 125
+    >>> assert (atg[20, 2000].weight == 195).all()
+    >>> assert isinstance(atg[0, 1990], pd.DataFrame)
+
+    If the column has the same name as a function (mean), then access it
+    with getitem,
+
+    >>> atg[:, :]["mean"] = [5.9]
+
+    """
+    def __init__(self, age_time_grid, columns, count=1):
+        assert isinstance(columns[0], str)
```

Comment:
If you're going to do it, do all the columns not just one

Reply:
Oh, I see, you're also allowing this to be a single string... I think I would group this in with the stuff on line 46 for clarity. I know that moves the precondition check away from the top of the function but I think that's ok...

Reply:
I moved these around to check things once. Thanks for the feedback.